token-handler/src/lib.rs

```diff
@@ -5,19 +5,27 @@ use multiversx_sc::imports::*;
 
 pub mod burn_tokens;
 pub mod common;
-pub mod events;
 pub mod mint_tokens;
 pub mod token_handler_proxy;
 
 #[multiversx_sc::contract]
 pub trait TokenHandler:
-    mint_tokens::MintTokens
-    + burn_tokens::BurnTokens
+    mint_tokens::MintTokensModule
+    + burn_tokens::BurnTokensModule
     + utils::UtilsModule
     + common::storage::CommonStorage
+    + tx_batch_module::TxBatchModule
+    + common::events::EventsModule
 {
     #[init]
-    fn init(&self, chain_prefix: ManagedBuffer) {
+    fn init(&self, header_verifier_address: ManagedAddress, chain_prefix: ManagedBuffer) {
```

Comment:
The 2 arguments are wrong.
I guess you need header_verifier_address this to call remove_executed_hash but this is wrong because 1) there are multiple header verifier contracts (one for each enshrine) and 2) remove_executed_hash has a condition to be called only by one specific enshrine.
Same problem for chain_prefix, there are multiple enshrine contracts that call one token-handler contract (in each shard)

Reply:
i think a good idea is to create a storage mapper dictionary so we have the associated addresses:
HeaderVerifier A <-> Enshrine A

Reply:
I also think a storage mapper dictionary is needed.

Separation of contracts is a must, so every enshrined sovereign must have its own set of contracts.

Reply:
- Yes, each Enshrine has a HeaderVerifier storage inside. TokenHandler only needs a list of Enshrine contracts, which is implemented [here](https://github.com/multiversx/mx-sovereign-sc/pull/76).
- TokenHandler is not connected to one Enshrine (to one prefix), is connected to all Enshrine from its shard. TokenHandler can mint any token with any prefix